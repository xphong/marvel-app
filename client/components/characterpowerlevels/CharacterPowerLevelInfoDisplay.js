import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class CharacterPowerLevelInfoDisplay extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const {character} = this.props;

    return (
      <div>
        <h1>{character.Name}</h1>
        <h4>Average Power Level: {character.AveragePowerLevel}</h4>
        <a target='_blank' href={character.Profile_Link}>
          <img className='powerlevel-image fadeIn-animation' src={character.Image_Link} />
        </a>
      </div>
    );
  }
}

CharacterPowerLevelInfoDisplay.propTypes = {
  character: PropTypes.object.isRequired
};
