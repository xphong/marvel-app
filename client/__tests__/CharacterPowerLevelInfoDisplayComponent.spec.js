import React from 'react';

import CharacterPowerLevelInfoDisplay from '../components/characterpowerlevels/CharacterPowerLevelInfoDisplay';

describe('CharacterPowerLevelInfoDisplay Component', () => {
  it('should render CharacterPowerLevelInfoDisplay', () => {
    const props = {
      character: {
        Name: 'Rogue',
        AveragePowerLevel: '9001'
      }
    };

    const wrapper = shallow(<CharacterPowerLevelInfoDisplay {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
