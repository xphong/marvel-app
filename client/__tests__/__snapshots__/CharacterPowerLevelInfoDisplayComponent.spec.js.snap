// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPowerLevelInfoDisplay Component should render CharacterPowerLevelInfoDisplay 1`] = `
<div>
  <h1>
    Rogue
  </h1>
  <h4>
    Average Power Level: 
    9001
  </h4>
  <a
    target="_blank"
  >
    <img
      className="powerlevel-image fadeIn-animation"
    />
  </a>
</div>
`;
